src/inference/inference_engine.py

```diff
@@ -24,6 +24,22 @@ def estimate_pose_keypoints(self) -> Dict[str, List[VideoPoseResult]]:
                 try:
                     video_pose_result = estimator.estimate_pose(video.path)
                     results[estimator.name].append(video_pose_result)
+                    
+                    video_name, _ = os.path.splitext(video.path) # remove extension
+                    json_folderpath = os.path.join(self.base_output_path, os.path.basename(video_name))
+                    os.makedirs(json_folderpath, exist_ok=True)
+                    
+                    json_filepath = os.path.join(json_folderpath, f"{estimator.name}.json")
+                    serialized_video_pose_result = {
+                        "fps": video_pose_result.fps,
+                        "frame_width": video_pose_result.frame_width,
+                        "frame_height": video_pose_result.frame_height,
+                        "frames": [asdict(frame) for frame in video_pose_result.frames]
+                    }
+                    
+                    with open(json_filepath, "w+") as f:
+                        json.dump(serialized_video_pose_result, f, indent=2)
```

Comment:
Please extract this codeblock to a separate method like `save_predicted_pose_result` to have a clean interface and make the run method of the inference engine easy to understand.